docs/source/guides/character_control_guide.rst

```diff
@@ -0,0 +1,159 @@
+================================
+NuCypher Character Control Guide
+================================
+
+The `Character Control` is a module that contains useful REST-like HTTP endpoints for working with NuCypher characters.
+
+.. important::
+
+   Character control is currently a Work-In-Progress. Expect large, and even breaking, changes to this API often.
+
+.. warning::
+
+    Character control is currently not intended for use over remote connections or on shared machines.
+    The current API has not been secured and should not be used for production applications.
+
+.. contents:: Table of Contents
+   :depth: 4
+
+
+API Request/Response Structure Overview
+=======================================
+
+Status Codes
+------------
+All documented API endpoints use JSON and are REST-like.
+
+Some common returned status codes you may encounter are:
+
+- ``200 OK`` -- The request has succeeded.
+- ``400 BAD REQUEST`` -- The server cannot or will not process the request do to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).
+- ``500 INTERNAL SERVER ERROR`` -- The server encountered an unexpected condition that prevented it from fulfilling the request.
+
+Typically, you will want to ensure that any given response from a character control endpoint results in a 200 status code.
+This tells you that the server successfully completed the call.
+
+If you are returned a 400, check the data that you're sending to the server.
+Chances are you might be sending some invalid data to the server.
+See below for what the character control API expects.
+
+.. _`Contribution Guide`: https://docs.nucypher.com/en/latest/guides/contribution_guide.html/
+.. _`Discord`: https://discord.gg/7rmXa3S
+
+If you are ever given a 500 status code, we'd like to know about it!
+You can see our `Contribution Guide`_ for getting involved.
+Ideally, you can share some information with us about what you were doing when you encountered the 500 in the form of a GitHub issue, or just tell us in our `Discord`_.
+
+HTTP Methods
+------------
+Currently, the character control API only uses the following HTTP methods:
+
+- ``POST``
+- ``PUT``
+
+We don't exactly follow RESTful methodology precisely.
+Take careful note following the API endpoints to see what to send and what to expect as a response.
+
+Request Format
+--------------
+The character control API uses JSON for all its endpoints. A request may look like:
+
+.. code::
+
+    {
+        'bob_encrypting_key': '0324df67664e6ea40f2eea8037c994debd4caa42117fe86cdb8cab6ac7728751ad',
+        'label': 'dGVzdA==',
+        'm': 2,
+        'n': 3,
+        'expiration_time': '2019-02-14T22:23:10.771093Z',
+    }
+
+Take a look at ``bob_encrypting_key``. Take note that it's a hex-encoded string.
+The character control API endpoints expect `all` keys to be encoded as hex.
+
+Now, look at ``label``. Notice that it's a base64-encoded string.
```

Comment:
Interesting. So, we wouldn't support using ASCII labels like `"my_label"` in the API to avoid ambiguity and use `base64` only?

Reply:
I think using base64 is wise, since there may be use cases where the label is a bytes sequence.

Reply:
It's only base64 as a serialization layer over the wire since it can be any bytestring.